neo/Ledger/PoolItem.cs

```diff
@@ -0,0 +1,57 @@
+using Neo.Network.P2P.Payloads;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Akka.Util.Internal;
+using Neo.Network.P2P;
+using Neo.Persistence;
+using Neo.Plugins;
+
+namespace Neo.Ledger
+{
+    internal class PoolItem : IComparable<PoolItem>
+    {
+        /// <summary>
+        /// Internal transaction for PoolItem
+        /// </summary>
+        public readonly Transaction Tx;
+
+        /// <summary>
+        /// Timestamp when transaction was stored on PoolItem
+        /// </summary>
+        public readonly DateTime Timestamp;
+
+        /// <summary>
+        /// Timestamp where this transaction was last broadcast to other nodes
+        /// </summary>
+        public DateTime LastBroadcastTimestamp;
+
+        internal PoolItem(Transaction tx)
+        {
+            Tx = tx;
+            Timestamp = TimeProvider.Current.UtcNow;
+            LastBroadcastTimestamp = Timestamp;
+        }
+
+        public int CompareTo(Transaction otherTx)
+        {
+            if (otherTx == null) return 1;
+            // Fees sorted ascending
```

Comment:
will need to sort first here by whether the Tx is one of the `HigherLowPriorityTxTypes`

Reply:
For the PoolItems created in the high priority sorted set, you can just pass null for HigherLowPriorityTxTypes so that it doesn't sort on anything first for that case.

Reply:
Since plugin is doing the final sorting, so it's ok by now. We need to make sure that the decisions made in this class are correct, independent of plugins and other components.

Reply:
I'm already finishing basic tests. 5 minutes

Reply:
This independence was very important step, brother, opened several discussions and improvements.
For the plugins it was opened here: https://github.com/neo-project/neo-plugins/pull/39

Reply:
@igormcoelho, what to do here?
Should we add a comparison for high and low priority, such as:
` int ret = (!Tx.IsLowPriority).CompareTo(!otherTx.IsLowPriority);
if (ret != 0) return ret;`

Reply:
It will be kind of pointless comparison because two poolitems of different Class are not compared.

Reply:
If we need High/Low priority decisions here, perhaps it's better to adopt @erikzhang suggestion months ago, that was to put that directly on ClaimTransaction, not on plugins.

Reply:
yeah, it really is better not to be coming from plugins in my opinion also. Are there any complications though in moving it to ClaimTransaction in terms of compatibility?

Reply:
two pool items of a different priority are never in the same sorted set in the MemoryPool, so I don't think it needs to compare based on high or low priority of the transaction.

Reply:
Since we are now exposing it to the rest of the project though, maybe we should include sorting by priority inside of it just in case?

@igormcoelho what do you think the changes are of it being abused in the future if we don't add sorting by priority inside this now?

Reply:
Lets just add it. so it will check by priority first, then if it is a low priority transaction check whether it is one of the `HigherLowPriorityTxTypes`.

Reply:
What are the costs of additional checks like this, in practice? 
I mean, if we consider something like a mempool with 100k or 1M TXs?

Reply:
I was mentioning this specific comment here: https://github.com/neo-project/neo/pull/510#discussion_r240027402
But my point is that there's no need to take in account High/Low priority decisions inside PoolItem, because it's already separated by MemoryPool. If it gets too complicated, perhaps we can move this class back to MemoryPool and keep it `internal`.

Reply:
I'll resolve this conversation; we can have a separate PR for sorting first by whether the Tx is a claim transaction.